subprojects/shipkit/src/main/groovy/org/shipkit/internal/util/GitHubStatusCheck.java

```diff
@@ -103,8 +93,16 @@ private JsonObject findFirstError(Collection<JsonObject> statuses) {
 
     private boolean hasErrorStates(JsonObject statusDetails) {
         return statusDetails != null &&
-               statusDetails.getString("state") != null &&
-               (statusDetails.getString("state").equals("error") ||
-               statusDetails.getString("state").equals("failure"));
+            statusDetails.getString("state") != null &&
+            (statusDetails.getString("state").equals("error") ||
+                statusDetails.getString("state").equals("failure"));
+    }
+
+    private void waitingMethod(Long t) {
```

Comment:
Why not pushing this to retryManager, making it testable? The more complexity we push out the better!

Reply:
I wanted to extract waiting method to be passed from external as a lambda expression for each usage of RetryManager as there might be different waiting approaches, but only one way of counting it which is done by RetryManager logic.

Reply:
I think so. This way we push out more complexity, simplifying code, and potentially making things easier to test. It is an internal API so we can always change it. Good question!